ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -187,36 +123,64 @@ public KsqlResource(
       final Duration distributedCmdResponseTimeout,
       final ActivenessRegistrar activenessRegistrar
   ) {
-    this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    this(
+        ksqlConfig,
+        ksqlEngine,
+        serviceContext,
+        commandQueue,
+        distributedCmdResponseTimeout,
+        activenessRegistrar,
+        new RequestValidator(
+            CUSTOM_VALIDATORS,
+            OPTIONAL_VALIDATION,
+            () -> new EngineSnapshot(ksqlEngine.createSandbox()),
```

Comment:
Originally I had this PR coupled in with functionality changes and this is an artifact of that. In the next PR, I will have EngineSnapshot have another field `Checksum` which represents at what point the snapshot was taken (i.e. the checksum will be the offset that it is currently at in the command topic). This checksum will then be piped into the `enqueueCommand` so that the executor can verify that there was no race.